config/samples/resources/alloydbinstance/read-instance/alloydb_v1beta1_alloydbinstance.yaml

```diff
@@ -37,4 +37,18 @@ spec:
   machineConfig:
     cpuCount: 2
   readPoolConfig:
-    nodeCount: 3
\ No newline at end of file
+    nodeCount: 3
+  queryInsightsConfig:
```

Comment:
Have you verified that the boolean fields are set properly (indeed set to `false`/`true`) in the underlying GCP resource when applying this manifests? You can try it by building KCC locally and applying the manifest(s). Here is a similar guide: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/CONTRIBUTING.md#set-up-your-dev-environment 

The reason I asked is that setting a primitive-typed field to its empty value (`false`/`0`/`""`) in the KCC resource will be treated as unsetting the field when talking to the underlying GCP resource. Some more context: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2784#discussion_r1785307536